### PR TITLE
#10 Refactor user management into the cluster custom resource. Refact…

### PR DESCRIPTION
**Pull request by bfink13**
…or other classes into various packages.

### CODE REVIEW

File: src/main/java/com/indeed/operators/rabbitmq/api/RabbitMQApiClient.java [hunk 1]
```diff
@@ -163,8 +161,8 @@ private String buildAuthorizationHeader(final RabbitMQConnectionInfo connectionI
 
         Preconditions.checkNotNull(secret, String.format("Could not find secret with name [%s] in namespace [%s]", secretName, connectionInfo.getNamespace()));
 
-        final String username = secretsController.decodeSecretPayload(secret.getData().get("default-username"));
-        final String password = secretsController.decodeSecretPayload(secret.getData().get("default-password"));
+        final String username = secretsController.decodeSecretPayload(secret.getData().get("username"));
```
Comment (bfink13) [inline src/main/java/com/indeed/operators/rabbitmq/api/RabbitMQApiClient.java]: the fact that we ever used "default-username" and "default-password" was a little silly :)

File: src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ClusterUsersReconciler.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    public void reconcile(final RabbitMQCluster cluster) {
+        final RabbitMQConnectionInfo connectionInfo = new RabbitMQConnectionInfo(
+                cluster.getName(),
+                cluster.getNamespace(),
+                RabbitMQServices.getDiscoveryServiceName(cluster.getName())
+        );
+
+        final Map<String, RabbitMQUser> expectedUsers = cluster.getUsers().stream().collect(Collectors.toMap(RabbitMQUser::getUsername, user -> user));
```
Comment (bfink13) [inline src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ClusterUsersReconciler.java:49]: yes, i realize this whole method is pretty disgusting. i'll refactor it in the future.

File: src/main/java/com/indeed/operators/rabbitmq/resources/RabbitMQSecrets.java [hunk 3]
```diff
@@ -56,31 +56,31 @@ public Secret createClusterSecret(final RabbitMQCustomResource rabbit) {
...
     }
 
-    public Secret createUserSecret(final RabbitMQUserCustomResource user) {
-        final String username = user.getName();
-        final RabbitMQUserCustomResourceSpec spec = user.getSpec();
-        final String clusterName = spec.getClusterName();
-
+    public Secret createUserSecret(
+            final String username,
+            final RabbitMQCustomResource rabbit
```
Comment (bfink13) [inline src/main/java/com/indeed/operators/rabbitmq/resources/RabbitMQSecrets.java:62]: a bit unfortunate we have to pass this, but i couldn't find an alternative.

File: src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java [hunk 4]
```diff
@@ -39,8 +39,8 @@ public void reconcile(final RabbitMQCluster cluster) {
 
             Preconditions.checkNotNull(secret, String.format("Could not find secret with name [%s] in namespace [%s]", destSecretName, destSecretNamespace));
 
-            final String username = secretsController.decodeSecretPayload(secret.getData().get("default-username"));
-            final String password = secretsController.decodeSecretPayload(secret.getData().get("default-password"));
+            final String username = secretsController.decodeSecretPayload(secret.getData().get("username"));
```
Comment (jhamacher-indeed) [inline src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: I haven't really been paying attention but it seems like I've seen these changes at least three or four times already.  Should these be constants?

Comment (jhamacher-indeed) [reply src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: "username" and "password", I mean.

**Review by jhamacher-indeed — APPROVED** [review]
Generally LGTM.  I really like this change.

File: src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java (hunk 4 above)
Comment (bfink13) [reply src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: yea probably. stupid laziness!

Comment (jhamacher-indeed) [reply src/main/java/com/indeed/operators/rabbitmq/reconciliation/rabbitmq/ShovelReconciler.java]: Thanks!

**Pull request merged**